smartsim/_core/mli/infrastructure/control/workermanager.py

```diff
@@ -240,41 +260,85 @@ def _on_iteration(self) -> None:
         # # let the worker perform additional custom deserialization
         # request = self._worker.deserialize(request_bytes)
 
-        fetch_model_result = self._worker.fetch_model(request, self._feature_store)
-        model_result = self._worker.load_model(request, fetch_model_result)
-        fetch_input_result = self._worker.fetch_inputs(request, self._feature_store)
-        transformed_input = self._worker.transform_input(request, fetch_input_result)
-
-        # batch: t.Collection[_Datum] = transform_result.transformed_input
-        # if self._batch_size:
-        #     batch = self._worker.batch_requests(transform_result, self._batch_size)
-
         reply = InferenceReply()
 
+        fetch_model_result = None
+        model_result = None
+        fetch_input_result = None
+        transformed_input = None
+        execute_result = None
+        transformed_output = None
+
         try:
-            execute_result = self._worker.execute(
-                request, model_result, transformed_input
-            )
+            fetch_model_result = self._worker.fetch_model(request, self._feature_store)
+        except Exception as e:
```

Comment:
can we reduce noise and repetition here by passing the action to the exception handler?

```py

def fail_handler(
    func: t.Callable[..., t.Any],  # <-----
    func_args: t.Tuple[t.Any, ...],  # <-----
    exc: Exception,
    func_descriptor: str,
    reply: InferenceReply,
) -> None:
    ...
    try:
        return func(*func_args)  # <-----
    except:
        logger.exception(
            f"An error occurred while {func_descriptor}.\n"
            f"Exception type: {type(exc).__name__}.\n"
            f"Exception message: {str(exc)}"
        )
        reply.status_enum = "fail"
        reply.message = f"Failed while {func_descriptor}."
```

and modifying the calls like:

```py
# fetch_model_result = self._worker.fetch_model(request, self._feature_store)
fetch_model_result = fail_handler(
  self._worker.fetch_model,
  (self._feature_store, )
  "fetching the model",
  reply
)
```

Reply:
another idea here for reducing verbosity is to avoid all the extra checks....

```py
# send reply on fail
def exception_handler(
    exc: Exception, func_descriptor: str, reply: InferenceReply,
    reply_channel: CommChannelBase #<------
) -> None:
  ...
  reply.statu_enum = "failed"
  reply.message = f"failed while..."
  reply_channel.send(reply)
```

and return on exceptions

```py
try:
            fetch_model_result = self._worker.fetch_model(request, self._feature_store)
        except Exception as e:
            exception_handler(e, "fetching the model", reply)
            return. # <-----
```


```

Reply:
Oh I like the idea of sending the failed response asap!

Reply:
I went with the last option!